packages/Operators/test/tstVirtualWork.cpp

```diff
@@ -0,0 +1,285 @@
+    Teuchos::RCP<DataTransferKit::MapOperator> cloud_op =
+        factory.create( structure_displacements_vector->getMap(), 
+                        fluid_displacements_vector->getMap(), 
+                        *parameters );
+
+    // Setup the operator.
+    cloud_op->setup( structure_manager.functionSpace(),
+                     fluid_manager.functionSpace() );
+
+    // Map the displacements.
+    cloud_op->apply( *structure_displacements_vector, 
+                     *fluid_displacements_vector, 
+                     Teuchos::NO_TRANS );
+    
+    // Map the forces.
+    cloud_op->apply( *fluid_forces_vector, 
+                     *structure_forces_vector, 
+                     Teuchos::TRANS );
+
+    // Calculate structure virtual work.
+    double local_structure_virtual_work = 0.0;
+    for ( int i = 0; i < num_points; ++i )
+    {
+        for ( int d = 0; d < space_dim; ++d )
+        {
+            local_structure_virtual_work += 
+                structure_displacements[num_points*d+i] *
+                structure_forces[num_points*d+i];
+        }
+    }
+    Teuchos::reduceAll( *comm, 
+                        Teuchos::REDUCE_SUM,
+                        local_structure_virtual_work,
+                        Teuchos::ptrFromRef(structure_virtual_work) );
+
+    // Calculate fluid virtual work.
+    double local_fluid_virtual_work = 0.0;
+    for ( int i = 0; i < num_points; ++i )
+    {
+        for ( int d = 0; d < space_dim; ++d )
+        {
+            local_fluid_virtual_work += 
+                fluid_displacements[num_points*d+i] *
+                fluid_forces[num_points*d+i];
+        }
+    }
+    Teuchos::reduceAll( *comm, 
+                        Teuchos::REDUCE_SUM,
+                        local_fluid_virtual_work,
+                        Teuchos::ptrFromRef(fluid_virtual_work) );
+}
+
+//---------------------------------------------------------------------------//
+// Tests.
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( VirtualWork, mls_radius_test )
+{
+    // Run the test.
+    double structure_virtual_work = 0.0;
+    double fluid_virtual_work = 0.0;
+    setupAndRunTest(
+	"mls_test_radius.xml", structure_virtual_work, fluid_virtual_work );
+    
+    // Check the results.
+    TEST_FLOATING_EQUALITY( fluid_virtual_work, 
+                            structure_virtual_work, 
+                            epsilon );
+}
+
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( VirtualWork, mls_knn_test )
+{
+    // Run the test.
+    double structure_virtual_work = 0.0;
+    double fluid_virtual_work = 0.0;
+    setupAndRunTest(
+	"mls_test_knn.xml", structure_virtual_work, fluid_virtual_work );
+    
+    // Check the results.
+    TEST_FLOATING_EQUALITY( fluid_virtual_work, 
+                            structure_virtual_work, 
+                            epsilon );
+}
```

Comment:
That's where I think Boost.Test is superior to the Teuchos Unit Testing harness with the [data-driven test cases](http://www.boost.org/doc/libs/1_61_0/libs/test/doc/html/boost_test/tests_organization/test_cases/test_case_generation.html).